presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -338,6 +340,11 @@ public TestingPrestoServer(
         announcer = injector.getInstance(Announcer.class);
         requestBlocker = injector.getInstance(RequestBlocker.class);
 
+        // Announce Thrift server address
+        DriftServer driftServer = injector.getInstance(DriftServer.class);
+        driftServer.start();
+        updateThriftServerAddressAnnouncement(announcer, driftServerPort(driftServer), nodeManager);
```

Comment:
Do we need to add the similar thrift port update logic to `PrestoServer` as well ?

Reply:
It is in the following commit